### DISCUSSION TITLE
MTS200 monitor energy usage

### DISCUSSION THREAD

**Discussion by wiikka** [krahabb/meross_lan#484] [General]
Is there any way to monitor or estimate smart thermostat energy usage (E.g. MTS200)? E.g. Meross app has `Heating Time` in app, but is this possible to have as sensor? E.g. PowerCalc could be used to roughly estimate energy usage with heating time?

**Comment by krahabb** [comment]
Hello @wiikka,
Yep, it could be possible to add a 'calculated' sensor measuring the time the thermostat is active so that you can then feed this data to your energy estimation algorithm.
The point is, depending on the type of heater, the actual effective 'output power' is not known to the Meross device so this estimate could be very very inaccurate. At any rate...

I don't know when I could add this sensor. 
At the moment, you could anyway easily proceed this way to accomplish the same:
* Setup a template sensor based off the HVACAction of the climate entity. This numeric sensor should:
  - report `0` when HVACAction -> Idle
  - report `1`** when HVACAction -> Heating
   ** The value `1` will then be integrated (multiplied by the time base and added) over time and determine the 'scale' of the integral sum so you'll have to adjust this in order to have the integral sum provide the wanted 'time unit' (you could decide to count seconds, minutes, hours, etc..)
2. Then you'll have to use a [Riemann sum](https://www.home-assistant.io/integrations/integration/)  to 'integrate' the template sensor value so that you'll have the total 'Heating' time.

**Reply by wiikka** [reply]
Thanks! You're right the `HVACAction` is enough, no need for extra sensor for this case I'll guess.

I might be wrong, but I understood that electric underfloor heating is either on or off, and it uses same amount of power when on? And if you know how many square meters given circuit has, you could calculate the output power roughly

**Reply by krahabb** [reply]
Yep, if you have the mts200 for undeflow electric heating the power consumption of the heater should be 'almost' constant when `on` (it varies though, according to the temperature of the heater but that should be negligible)
The same is not true if the MTS200 is used to drive a gas boiler...
